chord_metadata_service/discovery/fields.py

```diff
@@ -91,16 +65,19 @@ async def get_distinct_field_values(field_props: DiscoveryFieldProps, low_counts
     # - e.g., if there are three individuals with sex=UNKNOWN_SEX, this
     #   should be treated as if the field isn't in the database at all.
 
-    model, field = get_model_and_field(field_props["mapping"])
+    model, field = f_utils.get_model_and_field(field_props["mapping"])
     threshold = get_threshold(low_counts_censored)
 
+    field_query = field
+    if group_by := field_props.get("group_by"):
+        # JSONField containing an array
+        # use jsonb_path_query field expression
+        field_query = f_utils.get_jsonb_path_query(field, group_by)
```

Comment:
this branch is still untested; if it's annoying to do so then no worries though.